### DISCUSSION TITLE
Issues Enabling Chaos Monkey

### DISCUSSION THREAD

**Discussion by Ivan-Bobrov** [spring-petclinic/spring-petclinic-microservices#262] [Q&A] [answered]
Hello spring-petclinic-microservices community,

I've been trying to use Chaos Monkey to test fault tolerance in the **visits** service. However, I've encountered several issues in the process and would appreciate your insights or suggestions on how to resolve them.

Here's a summary of the steps I've taken and the problems encountered:

1. I set the active profile for the visits service to `chaos-monkey` using IntelliJ IDEA's running configurations, aiming to enable the corresponding configurations from the `spring-petclinic-microservices-config` repository.
2. I started all the services locally **without** Docker, following the instructions provided in the README.
3. Attempting to run the `call_chaos.sh` script using **Git Bash** resulted in an error:

    ```text
    $ ./scripts/chaos/call_chaos.sh
    usage: ./scripts/chaos/call_chaos.sh: <customers|visits|vets> <attacks_enable_exception|attacks_enable_killapplication|attacks_enable_latency|attacks_enable_memory|watcher_enable_component|watcher_enable_controller|watcher_enable_repository|watcher_enable_restcontroller|watcher_enable_service|watcher_disable>
    First pick either customers, visits or vets
    Then pick what to enable. Order matters!
    Example
    ./scripts/chaos/call_chaos.sh visits attacks_enable_exception watcher_enable_restcontroller

    $ ./scripts/chaos/call_chaos.sh visits attacks_enable_exception watcher_enable_restcontroller
    curl: (7) Failed to connect to localhost port 8082 after 2233 ms: Couldn't connect to server
    ```

4. I then set the port for the visits service explicitly in the script to match the actual port it was running on:

    ```bash
    # Some code...
    while [[ $# > 0 ]]
    do
    key="$1"
    case $1 in
        # Some code...
        visits)
            PORT=5730 # Initially, it was 8082
            ;;
        # Some code...
    esac
    # Some code...
    ```

5. Running the script again using Git Bash led to another error:

    ```text
    $ ./scripts/chaos/call_chaos.sh visits attacks_enable_exception watcher_enable_restcontroller
    curl: (22) The requested URL returned error: 404
    ```

6. Upon checking, I noticed that the **Spring Actuator** for the visits service does **not** expose the `/chaosmonkey/assaults` endpoint.
7. In **Spring Boot Admin**, the environment for the visits service **correctly** shows as set to `chaos-monkey`.
8. However, in **Spring Boot Admin**, I do not see any configuration properties related to Chaos Monkey for the visits service.

Given these steps and the issues encountered, could you advise on what might be causing these problems?

**Comment by arey** [comment, marked as answer]
Hallo @Ivan-Bobrov 
It looks like the `chaos-monkey` dependency is old. I've done some tests and the chaos monkey actuator endpoint is not reachable `http://192.168.1.48:58647/actuator/chaosmonkey`
Could you please try with the last one:
```
 <chaos-monkey-spring-boot.version>3.1.0</chaos-monkey-spring-boot.version>
```

**Reply by Ivan-Bobrov** [reply]
Hi @arey,

Thank you for your help! Updating to the latest version of Chaos Monkey solved my issue. The Chaos Monkey actuator endpoint is now reachable, and everything is working perfectly.

**Reply by arey** [reply]
Thanks for the test. I will merge the PR
